artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonAbstractReceiver.java

```diff
@@ -46,9 +46,9 @@ public abstract class ProtonAbstractReceiver extends ProtonInitializable impleme
 
    // Cached instances used for this receiver which will be swapped as message of varying types
    // are sent to this receiver from the remote peer.
-   protected final MessageReader standardMessageReader = new AMQPMessageReader(this);
+   protected final MessageReader standardMessageReader;
```

Comment:
Same comment as above, don't change this, just use the accessor in the session context